### PR TITLE
[Term Entry] SwiftUI View Modifier: .background() #1510

### LINKED ISSUE #1510: [Term Entry] SwiftUI View Modifier: `.background()`
### Reason/inspiration (optional)

Please write a new entry under: content/swiftui/concepts/viewmodifier/terms/ about the modifier `.background()` in Swift.

A "Syntax" section with pseudo code snippet.
An "Example" section with a swift code snippet and image to accompany. Images should go in the media/ folder.

To get started with the PR and entry file:

Under content/swiftui/concepts/viewmodifier/terms/, create a new folder called `background`. Then create a new file called `background.md` under content/swiftui/concepts/viewmodifier/terms/`background`

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by matthewogtong**
### Description

- Created a new entry for the SwiftUI ViewModifier `.background()` method
Closes #1510

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The `.background()` modifier is then applied with the `Color.indigo` value passed in as the content parameter, to give the view an indigo background.
+
+Finally, a corner radius of 12 points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
+
+This will display:
+
+![SwiftUI ViewModifier Background](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-viewmodifier-background.png)
+
+> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If you want to apply a background to a group of views, you should consider using a container view, such as `ZStack`. Another tip is that you can use the `.opacity()` modifier along with `.background()` to set the transparency of the background.
```
Comment (SSwiniarski) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:65]: Can we rewrite the note to remove the second person pronoun? (You)

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@matthewogtong I did a first review and left one comment.

File: content/swiftui/concepts/viewmodifier/terms/background/background.md (hunk 1 above)
Comment (matthewogtong) [reply content/swiftui/concepts/viewmodifier/terms/background/background.md]: Thanks for the review and feedback! I have updated the note to remove second person pronouns.

**Review by SSwiniarski — APPROVED** [review]
@matthewogtong looks good for a second review.

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+```swift
+var body: some View {
+        VStack {
+            Text("I have an Indigo background!")
+        }
+        .padding()
+        .foregroundColor(.white)
+        .background(Color.indigo)
+        .cornerRadius(12)
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:51]:
```suggestion
    VStack {
        Text("I have an Indigo background!")
    }
    .padding()
    .foregroundColor(.white)
    .background(Color.indigo)
    .cornerRadius(12)
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## Syntax
+
+```pseudo
+View
+    .background(
+        _ content: View,
+        alignment: Alignment = .center,
+        blendMode: BlendMode? = nil,
+        ignoresSafeAreaEdges: Edge.Set = .init()
+    )
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:27]:
```suggestion
    .background(
        _ content: View,
        alignment: Alignment = .center,
        blendMode: BlendMode? = nil,
        ignoresSafeAreaEdges: Edge.Set = .init()
    )
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+This method has one required parameter:
+
+- `content`: The view to use as the background.
+
+And a few optional parameters:
+
+- `alignment`: The alignment of the background within the view.
+  > **Note**: The `alignment` parameter's default value is `center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:37]:
```suggestion
  > **Note**: The `alignment` parameter's default value is `.center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        blendMode: BlendMode? = nil,
+        ignoresSafeAreaEdges: Edge.Set = .init()
+    )
+```
+
+This method has one required parameter:
+
+- `content`: The view to use as the background.
+
+And a few optional parameters:
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:34]:
```suggestion
Additionally, there are a few optional parameters:
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+- `alignment`: The alignment of the background within the view.
+  > **Note**: The `alignment` parameter's default value is `center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
+- `blendMode`: The blend mode to use when rendering the background.
+- `ignoresSafeAreaEdges`: Whether the background should ignore safe area edges.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:39]:
```suggestion
- `ignoresSafeAreaEdges`: A boolean indicating whether the background should ignore safe area edges.
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            Text("I have an Indigo background!")
+        }
+        .padding()
+        .foregroundColor(.white)
+        .background(Color.indigo)
+        .cornerRadius(12)
+}
+```
+
+In the above example, we have a `Text` view displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:55]:
```suggestion
In the above example, a `Text` view ia displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+The `.background()` modifier is then applied with the `Color.indigo` value passed in as the content parameter, to give the view an indigo background.
+
+Finally, a corner radius of 12 points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
+
+This will display:
+
+![SwiftUI ViewModifier Background](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-viewmodifier-background.png)
+
+> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If a background is required for a group of views, it is recommended to use a container view, such as `ZStack`. Additionaly, the `.opacity()` modifier can be used along with `.background()` to adjust the transparency of the background.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:65]:
```suggestion
> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If a background is required for a group of views, it is recommended to use a container view, such as [`ZStack`](https://www.codecademy.com/resources/docs/swiftui/views/zstack). Additionaly, the `.opacity()` modifier can be used along with `.background()` to adjust the transparency of the background.
```

**Review by yangc95 — COMMENTED** [review]
@matthewogtong I did a second review. Please tag me once it's complete, thanks!

File: content/swiftui/concepts/viewmodifier/terms/background/background.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (matthewogtong) [reply content/swiftui/concepts/viewmodifier/terms/background/background.md]: Done! I've added possible values for those parameters.

File: content/swiftui/concepts/viewmodifier/terms/background/background.md (hunk 3 above)
Comment (matthewogtong) [reply content/swiftui/concepts/viewmodifier/terms/background/background.md:27]: Hi @yangc95, could you please clarify these suggested changes, I do not see any differences here.

**Comment by matthewogtong** [comment]
> @matthewogtong I did a second review. Please tag me once it's complete, thanks!

@yangc95 Thanks for the second review! I've addressed all the comments and feedback. Please let me know if there are any further changes I should make.

File: content/swiftui/concepts/viewmodifier/terms/background/background.md (hunk 3 above)
Comment (yangc95) [reply content/swiftui/concepts/viewmodifier/terms/background/background.md:27]: Please disregard this comment, the 4-spaces are correct

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+The **`.background()`** view modifier in SwiftUI can be used to apply a background to a view. This can be useful when creating custom views, or when trying to achieve a specific visual effect.
+
+## Syntax
+
+```pseudo
+View
+    .background(
+        _ content: View,
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:23]:
```suggestion
        content,
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        alignment: Alignment = .center,
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:24]:
```suggestion
        alignment: .alignment,
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        Content,
+        alignment: .center,
+        blendMode: .overlay,
+        ignoresSafeAreaEdges: .all
+    )
+```
+
+This method has one required parameter:
+
+- `content`: The view to use as the background.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:32]:
```suggestion
- `Content`: The view to use as the background.
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        Text("I have an Indigo background!")
+    }
+    .padding()
+    .foregroundColor(.white)
+    .background(Color.indigo)
+    .cornerRadius(12)
+}
+```
+
+In the above example, a `Text` view ia displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:57]:
```suggestion
In the above example, a `Text` view is displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```

File: content/swiftui/concepts/viewmodifier/terms/background/background.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+The `.background()` modifier is then applied with the `Color.indigo` value passed in as the content parameter, to give the view an indigo background.
+
+Finally, a corner radius of 12 points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
```
Comment (yangc95) [inline content/swiftui/concepts/viewmodifier/terms/background/background.md:61]:
```suggestion
Finally, a corner radius of `12` points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
```

**Review by yangc95 — APPROVED** [review]

**Pull request merged**